content/blogs/newsletter-104.md

```diff
@@ -0,0 +1,114 @@
+
+Another important improvement for SMP is the kernel initialization in the AMD64 build architecture, which was revamped by Timo ([PR #6110](https://github.com/reactos/reactos/pull/6110)).
+This allows SMP to work properly.
+
+{{< gallery >}}
+{{< figure link="/img/blogs/newsletter-104/blender_smp.png" src="/img/blogs/newsletter-104/blender_smp.png" caption="Blender running in ReactOS 64-bit with 2 cores">}}
+{{< /gallery >}}
+
+Timo also helped fix some bugs to be able to compile ReactOS in the GCC 13 build environment.
+Much of the thanks for preparing ReactOS to build on GCC 13 goes to Simone Mario Lombardo (simonelombardo) for making ReactOS bootable.
+
+In the meantime, Justin Miller (The_DarkFire) is focusing on the x86 architecture of the SMP area.
+He worked on setting up application processor bootup ([PR #5879](https://github.com/reactos/reactos/pull/5879)) in x86 SMP HAL and is planning to further improve x86 SMP.
+On top of all of this experiments have begun to allow ReactOS to make use of the new SMP aware HAL.
+
+{{< gallery >}}
+{{< figure link="/img/blogs/newsletter-104/smp_x86.png" src="/img/blogs/newsletter-104/smp_x86.png" caption="Experimenting the 32-bit SMP build of ReactOS">}}
+{{< figure link="/img/blogs/newsletter-104/smp_cpuz.png" src="/img/blogs/newsletter-104/smp_cpuz.png" caption="ReactOS SMP 32-bit showing the benchmark results with CPU-Z">}}
+{{< /gallery >}}
+
+**It is worth noting that SMP is still heavily under development and there is no exact estimate of when the multi-core support will finally be ready, so stay tuned for more updates!**
+
+## User-mode WINE modules syncing
+Timo, Justin and several other contributors are making efforts to bring up-to-date some of the user-mode modules that we borrow from WINE.
+As you may know; ReactOS shares many components in user mode with WINE, although our shell is in many cases an exception to this the bulk of the usermode DLLs that provide APIs to applications are synced with the WINE project as base.
+
+ReactOS for many years have fallen short on updating these components as WINE has pushed further into the modern era.
+Updating these components means we greatly improve the stability and functionality of the components we sync, but also lets us look into the future as more NT6+ features and functionality get merged into the project.
+
+Some early experiments have shown dramatically improved application compatibility on the very early builds that come enabled with these features by default.
+
+{{< gallery >}}
+{{< figure link="/img/blogs/newsletter-104/python.png" src="/img/blogs/newsletter-104/python.png" caption="Running Python on ReactOS (this is used to run the Winesync python script)">}}
+{{< /gallery >}}
+
+The aim of this effort is to bring the improvements and fixes from WINE that help with the support of NT6+ software that otherwise wouldn't have worked.
+This also helps getting rid of hacks introduced.
+
+## Asynchronous network connection support
+TODO
```

Comment:
Regarding async network patch, it's being worked on in [PR #6349](https://github.com/reactos/reactos/pull/6349) . I've 
added async connect support for `WSPConnect` function in MSAFD (patch originally made by Peter Hater), fixed IOCTL buffers handling and fixed the leaks in MSAFD code. Now the code was working fine for web browsers without crashing or web lookup issue. However, the implementation still relies on `SleepEx` hacks (for APCs) and the performance still not as fast as Windows. I'll find a better solution for MSAFD APCs handling in the future PRs.